code/test/general.js

```diff
@@ -0,0 +1,2306 @@
+const TWO_ETH = unit.toWei('2', 'ether')
+const THREE_ETH = unit.toWei('3', 'ether')
+const FOUR_ETH = unit.toWei('4', 'ether')
+const INTERVAL = 30000 // 30 second Intervals
+const DURATION = INTERVAL * 2 * 60 * 24 * 1 // 1 day wallet duration
+// const SLOT_SIZE = 1 // 1 transaction per interval
+const EFFECTIVE_TIME = Math.floor(Date.now() / INTERVAL / 6) * INTERVAL * 6 - DURATION / 2
+
+contract('ONEWallet', (accounts) => {
+  // Wallets effective time is the current time minus half the duration (3 minutes ago)
+  let snapshotId
+  beforeEach(async function () {
+    snapshotId = await TestUtil.snapshot()
+    await TestUtil.init()
+  })
+  afterEach(async function () {
+    await TestUtil.revert(snapshotId)
+  })
+
+  // ====== TRACK ======
+  // Test tacking of an ERC20 token
+  // Expected result the token is now tracked
+  it('OPERATION 0 TRACK: must be able to track tokens', async () => {
+    // create wallets and token contracts used througout the tests
+    let alice = await TestUtil.makeWallet('TG-OP0-1', accounts[0], EFFECTIVE_TIME, DURATION)
+    let aliceOldState = await TestUtil.getONEWalletState(alice.wallet)
+    let aliceCurrentState = await TestUtil.getONEWalletState(alice.wallet)
+    const { testerc20 } = await TestUtil.makeTokens(accounts[0])
+    let aliceBalanceERC20
+    let aliceWalletBalanceERC20
+    // transfer ERC20 tokens from accounts[0] (which owns the tokens) to alices wallet
+    await testerc20.transfer(alice.wallet.address, 1000, { from: accounts[0] })
+    aliceCurrentState = await TestUtil.getONEWalletState(alice.wallet)
+    aliceBalanceERC20 = await testerc20.balanceOf(alice.wallet.address)
+    assert.equal(1000, aliceBalanceERC20, 'Transfer of 1000 ERC20 tokens to alice.wallet succesful')
+    aliceWalletBalanceERC20 = await alice.wallet.getBalance(ONEConstants.TokenType.ERC20, testerc20.address, 0)
+    assert.equal(1000, aliceWalletBalanceERC20, 'Transfer of 1000 ERC20 tokens to alice.wallet checked via wallet succesful')
+    console.log('Funded ERC20')
+
+    // Begin Tests
+    let testTime = Date.now()
+
+    testTime = await TestUtil.bumpTestTime(testTime, 60)
+    await TestUtil.transactionExecute(
+      {
+        ...ONEConstants.NullOperationParams, // Default all fields to Null values than override
+        wallet: alice,
+        operationType: ONEConstants.OperationType.TRACK,
+        tokenType: ONEConstants.TokenType.ERC20,
+        contractAddress: testerc20.address,
+        testTime
+      }
+    )
+    // Update alice current State
+    aliceCurrentState = await TestUtil.getONEWalletState(alice.wallet)
+    // Alice Items that have changed - nonce, lastOperationTime, commits, trackedTokens
+    // nonce
+    let nonce = await alice.wallet.getNonce()
+    assert.notEqual(nonce, aliceOldState.nonce, 'alice wallet.nonce should have been changed')
+    assert.equal(nonce.toNumber(), aliceOldState.nonce + 1, 'alice wallet.nonce should have been changed')
```

Comment:
just need one between 68,69

Reply:
Removed 68 and Refactored with a test that accepts 🙏